tests/conftest.py

```diff
@@ -1674,12 +1674,40 @@ def tmpvenv_active_inherit(deactive_venv, tmpdir: py.path.local) -> Iterator[env
     loader.unload_modules_for_path(venv.site_packages_dir)
 
 
+@pytest.fixture(scope="session")
+def create_local_package_index_factory():
+    """
+    A fixture that acts as a factory to create empty local pip package indexes
+    during a test session. Each call creates a new index in a temporary directory.
+    """
+
+    created_directories = []
+
+    def _create_local_package_index(index_name: str = "simple"):
+        """
+        Creates an empty pip index with the given index_name.
+        """
+        tmpdir = tempfile.mkdtemp()
+        created_directories.append(tmpdir)  # Keep track of the tempdir for cleanup
+        index_dir = os.path.join(tmpdir, index_name)
+        os.makedirs(index_dir)
+        dir2pi(argv=["dir2pi", str(tmpdir)])
+        return index_dir
```

Comment:
I don't understand the purpose of the `index_name` argument. As far as I can see, the `dir2pi` function always creates a sub-directory `simple` that contains the index, so matter what the value of the `index_name` argument is. I think we can remove it and hard-code the value `simple`, like in the original fixture.

Reply:
Indeed, but I still liked the fact that in the test, by reading the call you would directly see that those are different indexes for sure, without the need to look at the fixture. I changed the description of the fixture to emphasize that each call creates a different temp dir.

Reply:
Fine by me to keep the argument if you feel like it makes the code more readable. But we should make sure that the returned `index_dir` refers to the index created by the `dir2pi()` method (i.e. the "simple" directory).

Reply:
Also to debug the test it helps: while rewriting a test, I got this error: 
```Packages this-package-does-not-exist were not found in the given indexes. (Looking in indexes: /tmp/tmpw83luvxx/simple, /tmp/tmpvcvwlq50/simple, /tmp/tmp1cpkys1u/simple, /tmp/tmp4sbqgugm/simple, /tmp/tmppxit2vyy/simple, /tmp/tmpvcvwlq50/simple, /tmp/tmp1cpkys1u/simple)``` and it is really hard to know what index is missing/wrongly placed. 
I'll put it back and make sure to return the index_dir referes to the right index

Reply:
The `tempfile.mkdtemp()` method accepts a `prefix` argument. You could incorporate the `index_name` in the directory name like that, while still returning the correct path.